frontend/src/metabase/writeback/components/ActionViz/ActionForm.tsx

```diff
@@ -1,27 +1,83 @@
-import React from "react";
+import React, { useState } from "react";
 
 import type {
   WritebackQueryAction,
   WritebackParameter,
   OnSubmitActionForm,
+  ActionDashboardCard,
+  ParametersForActionExecution,
 } from "metabase-types/api";
+import { getFormTitle } from "metabase/writeback/components/ActionCreator/FormCreator";
 
 import ActionParametersInputForm from "../../containers/ActionParametersInputForm";
+import ActionParametersInputModal from "../../containers/ActionParametersInputModal";
+import ActionButtonView from "./ActionButtonView";
+
 import {
   DataAppPageFormWrapper,
   DataAppPageFormTitle,
 } from "./ActionForm.styled";
 
 interface ActionFormProps {
   onSubmit: OnSubmitActionForm;
+  dashcard: ActionDashboardCard;
   missingParameters: WritebackParameter[];
   action: WritebackQueryAction;
+  shouldDisplayButton: boolean;
 }
 
-function ActionForm({ onSubmit, missingParameters, action }: ActionFormProps) {
+function ActionForm({
+  onSubmit,
+  dashcard,
+  missingParameters,
+  action,
+  shouldDisplayButton,
+}: ActionFormProps) {
+  const [showModal, setShowModal] = useState(false);
```

Comment:
Subjectively `showModal` sounds like a function name, maybe something like `isModalVisible` would be more readable?